File: mathematics/fundamentals/even-odd-query.py
# Weekly Challenges - Week 5 > Even Odd Query
# Is the number odd or even?
#
# https://www.hackerrank.com/contests/w5/challenges/even-odd-query
#

input()
A = list(map(int, input().split()))

for _ in range(int(input())):
    x, y = map(int, input().split())
    if A[x - 1] % 2 == 1 or x > y or (x < y and A[x] == 0):
        print("Odd")
    else:
        print("Even")

File: mathematics/fundamentals/even-odd-query.cpp
// https://www.hackerrank.com/challenges/even-odd-query/problem

#include <iostream>
#include <vector>

using namespace std;

#pragma GCC diagnostic ignored "-Wsign-conversion"

int main()
{
    vector<int> a;
    int n, q, x, y, t;
    cin >> n;

    while (n--)
    {
        cin >> t;
        a.push_back(t);
    }

    cin >> q;
    while (q--)
    {
        cin >> x >> y;

        if (a[x - 1] % 2 == 1 || x > y || (x < y && a[x] == 0))
            cout << "Odd" << endl;
        else
            cout << "Even" << endl;
    }

    return 0;
}